src/model/backup.ts

```diff
@@ -620,3 +626,80 @@ export async function fetchBackupPassword(): Promise<null | BackupPassword> {
     return null;
   }
 }
+
+export async function getDeviceUUID(): Promise<string | null> {
+  if (IS_ANDROID) {
+    return null;
+  }
+
+  return new Promise(resolve => {
+    DeviceUUID.getUUID((error: unknown, uuid: string[]) => {
+      if (error) {
+        console.log('Got error when trying to get uuid from Native side');
+        resolve(null);
+      } else {
+        resolve(uuid[0]);
+      }
+    });
+  });
+}
+
+/**
+ * Checks if the identifier is the same as the one stored in localstorage
+ * The identifier can get out of sync in two instances:
+ * 1. when the user reinstalls the app
+ * 2. when the user migrates phones (we really only care about this instance)
+ *
+ * The goal here is to not allow them into the app if they have broken keychain data from a phone migration
+ *
+ * @returns a promise function to be ran after successful biometric authentication
+ */
+export async function checkIdentifierOnLaunch() {
+  // Unable to really persist things on Android, so let's just exit early...
+  if (IS_ANDROID) {
+    return;
+  }
+
+  try {
+    const uuid = await getDeviceUUID();
+    if (!uuid) {
+      throw new Error('Unable to retrieve identifier for vendor');
+    }
+
+    const currentIdentifier = identifier.get(['identifier']);
+    console.log('CurrentIdentifier: ', currentIdentifier);
+    // if we don't have a current identifier, set it and exit early
+    if (!currentIdentifier) {
+      identifier.set(['identifier'], uuid);
+      return;
+    }
+
+    // // if our identifiers match up, we can assume no reinstall/migration
+    // if (currentIdentifier === uuid) {
+    //   return;
+    // }
+
+    return new Promise(resolve => {
+      Navigation.handleAction(Routes.CHECK_IDENTIFIER_SCREEN, {
+        // Just a reinstall, let's update the identifer and send them back to the app
+        onSuccess: async () => {
+          identifier.set(['identifier'], uuid);
+          Navigation.handleAction(Routes.WALLET_SCREEN, {});
+          resolve(true); // Resolve the promise here
+        },
+        // Detected a phone migration, let's clear the app and send them to the welcome screen
+        onFailure: async () => {
+          // TODO: Is there more we need to wipe here?
+          await kc.clear();
```

Comment:
I think we can get away with only deleting `rainbowAddressKey` key from the public keychain. It's less destructive in case something goes wrong and has the same benefits